cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -53,11 +53,20 @@ func (kcm KindClusterManager) Create(c *config.UnmanagedClusterConfig) (*Kuberne
 	kindProvider := kindcluster.NewProvider()
 	clusterConfig := kindcluster.CreateWithKubeconfigPath(c.KubeconfigPath)
 
-	// TODO(stmcginnis): Determine what we need to do for kind configuration
-	parsedKindConfig, err := kindConfigFromClusterConfig(c)
-	if err != nil {
-		return nil, err
+	// If a user has provided something in the ProviderConfiguration,
+	// assume it is a full kind configuration and don't attempt to produce
+	// a configuration from the unmanaged-cluster config
+	var parsedKindConfig []byte
+	var err error
+	if c.ProviderConfiguration != "" {
```

Comment:
@jpmcb I wonder whether this `if` logic can be simplified.

For example, why not execute `[]byte(c.ProviderConfiguration)` no matter what? Even if `c.ProviderConfiguration == ""`. wouldn't the result be the same?

With that, this could probably go from `if else` -> `if`.

lmk if I'm missing something.

Reply:
quick example of what i'm asking about: https://go.dev/play/p/UWvKRyh5VWU

Reply:
My recommendation is:

```go
// If a user has provided something in the ProviderConfiguration,
// assume it is a full kind configuration and don't attempt to produce
// a configuration from the unmanaged-cluster config
var err error
parsedKindConfig := []byte(c.ProviderConfiguration)

// when kind config was empty, create a kind config from ClusterConfig settings
if len(parsedKindConfig) < 1 {
    parsedKindConfig, err = kindConfigFromClusterConfig(c)
    if err != nil {
        return nil, fmt.Errorf("failed to generate a viable kind config. error was: %s", err)
    }
}
```

lmk if this makes sense!